Index: lib/SPINE/Handler/Admin/File.pm
===================================================================
--- lib/SPINE/Handler/Admin/File.pm	(revision 5)
+++ lib/SPINE/Handler/Admin/File.pm	(revision 6)
@@ -99,6 +99,9 @@
   $filename =~ s/[;\`'!\?\@\|\"\*~<>\^\(\)\[\]\{\}\$\n\f\a\r\0\t\s]+//g; 
   $error = "";
   
+  $filename =~ /^.*[\/\\](.*)$/;
+  $filename = $1 if $1;
+
   my @params = $request->param;
   my @checkboxes = grep { /^check\d*$/ } sort { $a <=> $b } @params;
 
